### ISSUE TITLE
Select Python dependency management tool

### ISSUE THREAD

**Issue by lpsinger** [nasa-gcn/gcn.nasa.gov#1755]
We need to select a Python dependency management tool that allows us to transform a list of dependencies and version constraints into a concrete set of resolved dependency versions, similar to NPM's package.json/package-lock.json.

I am aware of several options here, although there may be others:

- [Pipenv](https://pipenv.pypa.io/en/latest/)
- [Poetry](https://python-poetry.org)
- [pip-tools](https://github.com/jazzband/pip-tools)

Our requirements are:
- It must be able to generate a requirements.txt file, because Architect needs that in order to vendor the dependencies for the Lambda functions.
- It must be [supported by Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem).

**Comment by jak574** [comment]
My brief thoughts on this is that if we need to be able to generate a requirements.txt, this means pipenv is probably a good solution, and poetry is not. pipenv I think has a similar Pipfile.lock file that is analogous to what you have in NPM.

**Comment by lpsinger** [comment]
It looks like pip-compile outputs a requirements.txt file directly, whereas with Pipenv that is a second step.

Perhaps it is worth a try comparing how long it takes to generate a lock file with each of these tools. Do you have an input requirements.txt file that represents your dependencies from #1742?

**Comment by jak574** [comment]
For pip-compile, the following `requirements.in` works for #1742 (I just pasted into the comment as github won't let me attach with the correct filename). I can do a `pyproject.toml` version if necessary. 

```
astropy==6.0.0
astropy-iers-data
architect-functions==1.0.0
antares-client==1.5.0
boto3==1.34.2
fastapi==0.105.0
numpy==1.26.2
mangum==0.17.0
black
mypy
pytest
pip-tools
``` 

Note I've only fixed the versions of the main top level dependencies of the API code here, not the support stuff (e.g. mypy, black).

**Comment by lpsinger** [comment]
Fixed by #1760.
